src/cpp/rtps/transport/UDPTransportInterface.cpp

```diff
@@ -698,6 +674,33 @@ bool UDPTransportInterface::fillUnicastLocator(
     return true;
 }
 
+void UDPTransportInterface::remove_known_network_interfaces(
+        const SendResourceList& sender_resource_list,
+        std::vector<IPFinder::info_IP>& locNames,
+        bool return_loopback)
+{
+    get_ips(locNames, return_loopback);
```

Comment:
I would rename this method to something like `get_unknown_network_interfaces`, and clear the locNames inside it, instead of forcing the caller to have it empty before calling the method.

```suggestion
void UDPTransportInterface::get_unknown_network_interfaces(
        const SendResourceList& sender_resource_list,
        std::vector<IPFinder::info_IP>& locNames,
        bool return_loopback)
{
    locNames.clear();
    get_ips(locNames, return_loopback);
```